test/Microsoft.DocAsCode.Build.SchemaDriven.Tests/MergeMarkdownFragmentsTest.cs

```diff
@@ -36,31 +41,232 @@ public MergeMarkdownFragmentsTest()
+        }
+
+        [Fact]
+        public void TestInvalidSpaceMissing()
+        {
+            // Arrange
+            var mdFile = CreateFile(
+                "Suppressions.yml.md",
+                @"#head_1_without_space
+
+## `summary`
+markdown content
+",
+                _inputFolder);
+
+            // Act
+            Logger.RegisterListener(_listener);
+            try
+            {
+                BuildDocument(_files);
+            }
+            finally
+            {
+                Logger.UnregisterListener(_listener);
+            }
+
+            var logs = _listener.Items;
+            var warningLogs = logs.Where(l => l.Code == WarningCodes.Overwrite.InvalidMarkdownFragments);
+            Assert.True(File.Exists(_rawModelFilePath));
+            var rawModel = JsonUtility.Deserialize<JObject>(_rawModelFilePath);
+            Assert.Single(warningLogs);
+            Assert.Equal("Unable to parse markdown fragments: Expect L1InlineCodeHeading", warningLogs.First().Message);
+            Assert.Equal("0", warningLogs.First().Line);
+            Assert.Null(rawModel["summary"]);
+        }
+
+        [Fact]
+        public void TestValidSpaceMissing()
+        {
+            // Arrange
+            var mdFile = CreateFile(
+                "Suppressions.yml.md",
+                @"# `management.azure.com.advisor.suppressions`
+
+## `summary`
+markdown content
+
+##head_3_without_space
+markdown content
+",
+                _inputFolder);
+
+            // Act
+            BuildDocument(_files);
+
+            Assert.True(File.Exists(_rawModelFilePath));
+            var rawModel = JsonUtility.Deserialize<JObject>(_rawModelFilePath);
+            Assert.Contains("##head_3_without_space", rawModel["summary"].ToString());
+        }
+
+        [Fact]
+        public void TestInvalidYaml()
+        {
+            // Arrange
+            var mdFile = CreateFile(
+                "Suppressions.yml.md",
+                @"# `management.azure.com.advisor.suppressions`
+```
+author:author_without_space
+```
+
+## `summary`
+markdown content
+",
+                _inputFolder);
+
+            // Act
+            Logger.RegisterListener(_listener);
+            try
+            {
+                BuildDocument(_files);
+            }
+            finally
+            {
+                Logger.UnregisterListener(_listener);
+            }
+
+            var logs = _listener.Items;
+            var warningLogs = logs.Where(l => l.Code == WarningCodes.Overwrite.InvalidMarkdownFragments);
+            Assert.True(File.Exists(_rawModelFilePath));
+            var rawModel = JsonUtility.Deserialize<JObject>(_rawModelFilePath);
+            Assert.Single(warningLogs);
+            Assert.Equal("Unable to parse markdown fragments: Encountered an invalid YAML code block in line 1: (Line: 1, Col: 1, Idx: 0) - (Line: 1, Col: 28, Idx: 27): Exception during deserialization", warningLogs.First().Message);
```

Comment:
>1 [](start = 109, length = 1)

This is the line from the start of the YAML code block. Add the offset to make it start from file's first line?